### ISSUE TITLE
Linux Service Collector Crash

### ISSUE THREAD

**Issue by goldstar611** [Microsoft/AttackSurfaceAnalyzer#106]
**Describe the bug**
The "ServiceCollector" crashes on Linux with a "Index was outside the bounds of the array" message.

**To Reproduce**
1. Install Ubuntu 16.04.6
2. Clone repo and install build deps
3. Build CLI + GUI from the published build guide
4. Start GUI and select all of the options
5. Notice that ServiceCollector never finishes because it crashed (messages are in the console)

**Expected behavior**
No crashes

**Screenshots**
N/A

**System Configuration (please complete the following information):**
 - OS: Ubuntu 16.04
 - OS Version: Ubuntu 16.04.6
 - Application Version: master
 - CLI or GUI: GUI

**Additional Context**
Relevant messages from the console
```
stdout: [12:30:45 DBG] ^/run/systemd/ caught /run/systemd/inhibit
stdout: [12:30:45 WRN] Unable to get access control for /run/udev/link.dvd: Path doesn't exist!
stdout: [12:30:45 WRN] Unable to get access control for /run/udev/link.cdrw: Path doesn't exist!
stdout: [12:30:45 INF] Completed FileSystemCollector in 00h:00m:12s:667ms
[12:30:45 DBG] 00:00:12.6670000
stdout: [12:30:45 INF] Completed: FileSystemCollector
stdout: [12:30:45 INF] Executing OpenPortCollector.
[12:30:45 DBG] Collecting open port information...
stdout: [12:30:45 DBG] ExecuteLinux()
stdout: [12:30:45 INF] Completed OpenPortCollector in 00h:00m:00s:084ms
stdout: [12:30:45 DBG] 00:00:00.0840000
stdout: [12:30:45 INF] Completed: OpenPortCollector
stdout: [12:30:45 INF] Executing ServiceCollector.
stdout: [12:30:45 ERR] Error collecting from ServiceCollector: Index was outside the bounds of the array.    at AttackSurfaceAnalyzer.Collectors.Service.ServiceCollector.Execute() in /home/user/Desktop/AttackSurfaceAnalyzer/Lib/Collectors/Service/ServiceCollector.cs:line 209
   at AttackSurfaceAnalyzer.Cli.AttackSurfaceAnalyzerCLI.RunCollectCommand(CollectCommandOptions opts) in /home/user/Desktop/AttackSurfaceAnalyzer/Cli/Program.cs:line 1131
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at AttackSurfaceAnalyzer.Collectors.Service.ServiceCollector.Execute() in /home/user/Desktop/AttackSurfaceAnalyzer/Lib/Collectors/Service/ServiceCollector.cs:line 209
   at AttackSurfaceAnalyzer.Cli.AttackSurfaceAnalyzerCLI.RunCollectCommand(CollectCommandOptions opts) in /home/user/Desktop/AttackSurfaceAnalyzer/Cli/Program.cs:line 1131
stdout: [12:30:45 INF] Completed: ServiceCollector
stdout: [12:30:45 INF] Executing UserAccountCollector.
stdout: [12:30:45 DBG] ExecuteLinux()
stdout: [12:30:45 INF] Completed UserAccountCollector in 00h:00m:00s:021ms
stdout: [12:30:45 DBG] 00:00:00.0210000
stdout: [12:30:45 INF] Completed: UserAccountCollector
stdout: [12:30:45 INF] Completed: CertificateCollector
```

**Comment by gfs** [comment]
I think this was fixed with #104.

**Comment by gfs** [comment]
Confirmed this seems to be working on my end.
